### ISSUE TITLE
awss3 integration uses AWS SDKv1, though AWS already suggest using SDKv2

### ISSUE THREAD

**Issue by Bryksin** [spring-cloud/spring-cloud-config#2038]
**Problem Description**
When attaching AWS S3 bucket as a backend for config service it attempts to initialize: `AwsS3EnvironmentRepositoryFactory`
which is still using `com.amazonaws.services.s3.AmazonS3ClientBuilder` which is AWS SDKv1
where AWS recommend's to use new SDKv2 with `software.amazon.awssdk.services.s3.S3ClientBuilder`
We are using some of SSM features and for that already imported SDKv2, but now we need to integrate two SDK's - v1 and v2
or rewrite everything SSM related to SDK1 - which sounds like an inappropriate decision.

**Proposed Solution**
Spring Cloud Config should start using AWS SDKv2 when accessing AWS resources

**Comment by ryanjbaxter** [comment]
Interested in contributing a PR?

**Comment by Bryksin** [comment]
Would love to, but not now, I already have several projects in parallel, just no time for another, though I will consider

**Comment by eddumelendez** [comment]
Hi @ryanjbaxter 
is it ok if we have 2 modules for aws, one for sdk v1 and other for sdk v2?

**Comment by ryanjbaxter** [comment]
Well we currently don't have a separate module...I am fine with removing sdk v1 in the main branch all together

**Comment by Bryksin** [comment]
I still remember about PR invitation 😃
Hope I will find time soon!

**Comment by hpoettker** [comment]
Spring Vault has released 3.0.0-M1 and switched to v2. If I'm not mistaken, this precludes the possibility to stay (exclusively) on v1 in 4.x due to the dependency on Spring Vault. Switching completely seems the best option to me.
